### ISSUE TITLE
setupPlayer, add and play is not working and throwing warning after TrackPlayer.destroy()

### ISSUE THREAD

**Issue by nisheetsun** [react-native-kit/react-native-track-player#597]
# Configuration

  React Native Environment Info:

>     System:
>       OS: macOS 10.14.3
>       CPU: (4) x64 Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz
>       Memory: 297.03 MB / 8.00 GB
>       Shell: 3.2.57 - /bin/bash
>     Binaries:
>       Node: 8.11.3 - ~/.nvm/versions/node/v8.11.3/bin/node
>       Yarn: 1.9.2 - /opt/local/bin/yarn
>       npm: 6.4.0 - ~/.nvm/versions/node/v8.11.3/bin/npm
>       Watchman: 4.9.0 - /usr/local/bin/watchman
>     SDKs:
>       Android SDK:
>         API Levels: 23, 25, 26, 27, 28
>         Build Tools: 23.0.1, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.3, 28.0.2, 28.0.3
>     IDEs:
>       Android Studio: 3.2 AI-181.5540.7.32.5056338
>       Xcode: /undefined - /usr/bin/xcodebuild
>     npmPackages:
>       react: 16.8.3 => 16.8.3 
>       react-native: 0.59.8 => 0.59.8 
>     npmGlobalPackages:
>       react-native-video: 3.2.1
>       react-native-workers: 0.3.1

> "react-native-track-player": "1.1.4"

# Issue
On click of remote-stop, I am doing TrackPlayer.destroy(). Players stops and push notification gets closes as expected but after that when I am doing setupPlayer, updateOptions, reset and play, it's throwing a warning as mentioned below.

_Possible Unhandled Promise Rejection(id: 1): Error: You attempted to set the key '()' with value "{id:"1111", "url":"https://drive.google.com/uc?...", title:"Longing", "artist":"David", artwork:"https://..."}" on an object that is meant to be immutable and has been frozen_

P.S. - I took the code from example folder of this repository. You can run that to reproduce the same issue.

# Code

```
TrackPlayer.addEventListener('remote-stop', () => {
    PlayerStore.stopped = true;
    TrackPlayer.destroy();
  })
```

```
  togglePlayback = async () => {
    const currentTrack = await TrackPlayer.getCurrentTrack();
    if (currentTrack == null) {
      if (PlayerStore.stopped == true) {
        TrackPlayer.setupPlayer();
        TrackPlayer.updateOptions({
          stopWithApp: true,
          capabilities: [
            TrackPlayer.CAPABILITY_PLAY,
            TrackPlayer.CAPABILITY_PAUSE,
            TrackPlayer.CAPABILITY_SKIP_TO_NEXT,
            TrackPlayer.CAPABILITY_SKIP_TO_PREVIOUS,
            TrackPlayer.CAPABILITY_STOP
          ],
          compactCapabilities: [
            TrackPlayer.CAPABILITY_PLAY,
            TrackPlayer.CAPABILITY_PAUSE
          ]
        });
        PlayerStore.stopped = false;
      }
      await TrackPlayer.reset();
      await TrackPlayer.add(playlistData);
      await TrackPlayer.play();
    } else {
      if (PlayerStore.playbackState === TrackPlayer.STATE_PAUSED) {
        await TrackPlayer.play();
      } else {
        await TrackPlayer.pause();
      }
    }
  }
```

**Comment by nisheetsun** [comment]
Thanks in advance.

**Comment by shiroze** [comment]
For mine its work this way : 

```
togglePlayback = async () => {
if(PlayerStore.playbackState === TrackPlayer.STATE_NONE) {
      await TrackPlayer.setupPlayer({
      waitForBuffer: true,
    });

    await TrackPlayer.updateOptions({
      stopWithApp: true,
      capabilities: [
        TrackPlayer.CAPABILITY_PLAY,
        TrackPlayer.CAPABILITY_PAUSE,
        TrackPlayer.CAPABILITY_STOP
      ],
      compactCapabilities: [
        TrackPlayer.CAPABILITY_PLAY,
        TrackPlayer.CAPABILITY_STOP
      ],
      notificationCapabilities: [
        TrackPlayer.CAPABILITY_PLAY,
        TrackPlayer.CAPABILITY_STOP
      ]
    });
}
....
```

Hope this meet your expectations and the code work fine.

**Comment by mynameisguy** [comment]
having the exact same issue.
also having #742 so might be related.

**Comment by curiousdustin** [comment]
Fix released in [1.2.4](https://github.com/react-native-kit/react-native-track-player/releases/tag/v1.2.4)
